flyteadmin/auth/authzserver/metadata_provider.go

```diff
@@ -107,3 +122,45 @@ func NewService(config *authConfig.Config) OAuth2MetadataProvider {
 		cfg: config,
 	}
 }
+
+func sendAndRetryHttpRequest(ctx context.Context, client *http.Client, url string, retryAttempts int, retryDelay time.Duration) (*http.Response, error) {
+	var response *http.Response
+	var err error
+	totalAttempts := retryAttempts + 1 // Add 1 for the case where 0 retryAttempts are specified
+
+	backoff := wait.Backoff{
+		Duration: retryDelay,
+		Steps:    totalAttempts,
+		Cap:      0,
+	}
+
+	// first func is to determine retriability of error
+	// second is to run logic
+	err = retry.OnError(backoff,
+		func(err error) bool { //
+			if grpcErr := status.Code(err); grpcErr == codes.Internal {
+				logger.Debugf(ctx, "Failed to get oauth metadata, going to retry. Err: %v", err)
+				return true
+			}
+			return false
+		}, func() error { // Send HTTP request
+			response, err = client.Get(url)
+			if err != nil {
+				return err
+			}
+			if response != nil && response.StatusCode >= http.StatusInternalServerError && response.StatusCode <= http.StatusNetworkAuthenticationRequired {
+				return flyteErrors.NewFlyteAdminError(codes.Internal, "Failed to get oauth metadata.")
```

Comment:
can we be more specific here and use a known error that we can compare against in line 141, something like

```
var retryableOauthMetadataError = errors.New("failed to get oauth metadata")
...
if response != nil && response.StatusCode >= http.StatusInternalServerError && response.StatusCode <= http.StatusNetworkAuthenticationRequired {
				return retryableOauthMetadataError
...
```

and then in line 141 check for equality against this designated error only

Reply:
Great idea, added in!